modules/management/kubectl-apply/variables.tf

```diff
@@ -0,0 +1,38 @@
+/**
+  * Copyright 2024 Google LLC
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *      http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+variable "project_id" {
+  description = "The project ID that hosts the gke cluster."
+  type        = string
+}
+
+variable "cluster_id" {
+  description = "An identifier for the gke cluster resource with format projects/<project_id>/locations/<region>/clusters/<name>."
+  type        = string
+  nullable    = false
+}
+
+variable "apply_manifests" {
+  description = "A list of manifests to apply to GKE cluster using kubectl."
+  type = list(object({
+    content           = optional(string, null)
+    source            = optional(string, null)
+    template_vars     = optional(any, null)
+    server_side_apply = optional(bool, false)
```

Comment:
Would there be ever a condition where `server_side_apply` and `wait_for_rollout` will be separate for each manifest ? In other words, can we extract it to a top level variable of it's own ?